core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -2142,13 +2143,21 @@ private static RelDataType deriveTypeMapFromEntries(SqlOperatorBinding opBinding
    * @see SqlStdOperatorTable#LN
    * @see SqlStdOperatorTable#LOG10
    */
-  @LibraryOperator(libraries = {BIG_QUERY})
+  @LibraryOperator(libraries = {BIG_QUERY, POSTGRESQL})
```

Comment:
PostgreSQL has different semantics for its log function.
If it's only 1 argument, PostgreSQL is the same as LOG10() and BigQuery is the same as LN().
If it's 2-argument, PostgreSQL has the base as the first argument and BigQuery has the base as the second argument.

See:
https://cloud.google.com/bigquery/docs/reference/standard-sql/mathematical_functions#log
https://www.postgresql.org/docs/9.1/functions-math.html
Mentioned in https://issues.apache.org/jira/browse/CALCITE-6312

Reply:
get, As you can see, I encountered difficulties in implementing the log10 and ln functions. I think that only by solving this problem, everything will be solved.